### DISCUSSION TITLE
Mesh shader supports

### DISCUSSION THREAD

**Discussion by JunkuiZhang** [gpuweb/gpuweb#4236] [Q&A] [answered]
As an avid user to the WebGPU ecosystem, I am particularly interested in the addition of Mesh Shader functionality. I would like to know if there have been any developments or updates regarding the implementation of Mesh Shader in WebGPU.

Additionally, I was wondering if there is any roadmap or plan in place for the inclusion of Mesh Shader support in WebGPU. It would be incredibly helpful to have an idea of the timeline or milestones associated with this feature's integration into the WebGPU specification.

Thanks in advance.

**Comment by Kangz** [comment, marked as answer]
See https://github.com/gpuweb/gpuweb/issues/3015 mesh shaders will be hard to add because they aren't really portable at the moment between all APIs so a pretty difficult investigation is needed. Plus not all browsers have WebGPU v1 implemented yet so it's a bit early for large additions like this (maybe we can start looking later this year)

**Reply by JunkuiZhang** [reply]
Thank you for taking the time to respond to my inquiry. Personally, even though it is called "WebGPU," I often prefer to use WebGPU as a native tool due to its versatility in developing applications as if I were using a native API. This makes WebGPU function as a platform layer of sorts. I believe many others share this sentiment and utilize WebGPU in a similar manner. Considering this, the addition of Mesh Shader is an exciting prospect.

While I understand that introducing significant additions like Mesh Shaders might be premature at this stage, I am pleased to hear that the team plans to further investigate this feature later in the year. Thank you once again for your response, and I eagerly await any updates regarding the progress of Mesh Shader support in WebGPU.

**Reply by munrocket** [reply]
@JunkuiZhang I guess mesh shaders is possible to recreate with compute pipeline.
